internal/pkg/cli/svc_package.go

```diff
@@ -396,9 +404,9 @@ func (o *packageSvcOpts) getTargetEnv() (*config.Environment, error) {
 	return o.targetEnv, nil
 }
 
-// RecommendActions is a no-op for this command.
+// RecommendActions suggests recommended actions before the packaged template is used for deployment.
 func (o *packageSvcOpts) RecommendActions() error {
-	return nil
+	return isManifestCompatibleWithEnvironment(o.appliedManifest.(manifest.WorkloadManifest), o.envName, o.envFeaturesDescriber)
```

Comment:
nit: What do you think of renaming this fn to be like this:
```suggestion
	return validateManifestCompatiblityWithEnv(o.appliedManifest.(manifest.WorkloadManifest), o.envName, o.envFeaturesDescriber)
```
Just because `is*` makes me think that the return type is a `bool`

Reply:
Also, should we actually error here? I think recommending them to run `env deploy` by logging to `stderr` is totally fair and printing the template to `stdout`. I just don't know if  **`svc`**`package` should fail if the template is incompatible w the env

```
err := isManifestCompatibleWithEnvironment(...)
if err != nil {
   log.Errorln(err)
}
return nil
```

Reply:
I was mainly concerned with the pipeline flow: if we don't exit with a non-zero code, the build stage will go through, and the deployment stage will follow. The deployment will then fail because of the incompatibility.

Do you think we can use a flag such as `copilot svc package --check-compatibility` to do this? Just like how we handle `--upload-assets`.

Reply:
Oh! That's smart, ok let's keep it as is.

It would be awesome if the feature just works without people having to update their buildspec.